src/App.tsx

```diff
@@ -1,14 +1,65 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { getTodos } from './api';
+import { Todo } from './types/Todo';
+import { TodosFilter } from './types/TodosFilter';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [selectedFilter, setSelectedFilter] = useState<TodosFilter>(TodosFilter.all);
+  const [query, setQuery] = useState('');
+  const [isLoading, setIsLoading] = useState(false);
+  const [modalIsShown, setModalIsShown] = useState(false);
+  const [todoModalId, setTodoModalId] = useState(0);
+
+  const filteredTodos = (todosToFilter: Todo[]) => {
+    return todosToFilter.filter((todo) => {
+      const titleMatch = todo.title
+        .toLowerCase()
+        .includes(query.trim().toLowerCase());
+
+      switch (selectedFilter) {
+        case 'active':
+          return titleMatch && !todo.completed;
+        case 'completed':
+          return titleMatch && todo.completed;
+        default:
+          return titleMatch;
+      }
+    });
+  };
+
+  useEffect(() => {
+    setIsLoading(true);
+
+    getTodos()
+      .then(todosList => {
+        setTodos(filteredTodos(todosList));
+      })
+      .catch((e) => {
+        throw new Error(e);
+      })
+      .finally(() => setIsLoading(false));
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [query, selectedFilter]);
```

Comment:
You don't need to load todos every time the `query` or `selectedFilter` changes. You need to load once, when the App loads for the first time. This what task descriptions says. You need to filter loaded todos if `query` or `selectedFilter`